src/ripple/protocol/impl/Indexes.cpp

```diff
@@ -141,102 +108,52 @@ getQuality (uint256 const& uBase)
 uint256
 getTicketIndex (AccountID const& account, std::uint32_t uSequence)
 {
-    return sha512Half(
-        std::uint16_t(spaceTicket),
-        account,
-        std::uint32_t(uSequence));
-}
-
-uint256
-getRippleStateIndex (AccountID const& a, AccountID const& b, Currency const& currency)
-{
-    if (a < b)
-        return sha512Half(
-            std::uint16_t(spaceRipple),
-            a,
-            b,
-            currency);
-    return sha512Half(
-        std::uint16_t(spaceRipple),
-        b,
-        a,
-        currency);
-}
-
-uint256
-getRippleStateIndex (AccountID const& a, Issue const& issue)
-{
-    return getRippleStateIndex (a, issue.account, issue.currency);
-}
-
-uint256
-getSignerListIndex (AccountID const& account)
-{
-    // We are prepared for there to be multiple SignerLists in the future,
-    // but we don't have them yet.  In anticipation of multiple SignerLists
-    // We supply a 32-bit ID to locate the SignerList.  Until we actually
-    // *have* multiple signer lists, we can default that ID to zero.
-    return sha512Half(
-        std::uint16_t(spaceSignerList),
-        account,
-        std::uint32_t (0));  // 0 == default SignerList ID.
-}
-
-uint256
-getCheckIndex (AccountID const& account, std::uint32_t uSequence)
-{
-    return sha512Half(
-        std::uint16_t(spaceCheck),
-        account,
-        std::uint32_t(uSequence));
-}
-
-uint256
-getDepositPreauthIndex (AccountID const& owner, AccountID const& preauthorized)
-{
-    return sha512Half(
-        std::uint16_t(spaceDepositPreauth),
-        owner,
-        preauthorized);
+    return indexHash(LedgerNameSpace::TICKET,
+        account, std::uint32_t(uSequence));
 }
 
 //------------------------------------------------------------------------------
 
 namespace keylet {
 
-Keylet account_t::operator()(
-    AccountID const& id) const
+Keylet account(
+    AccountID const& id) noexcept
 {
-    return { ltACCOUNT_ROOT, sha512Half(std::uint16_t(spaceAccount), id) };
+    return { ltACCOUNT_ROOT,
+        indexHash(LedgerNameSpace::ACCOUNT, id) };
 }
 
-Keylet child (uint256 const& key)
+Keylet child (uint256 const& key) noexcept
 {
     return { ltCHILD, key };
 }
 
-Keylet skip_t::operator()() const
+Keylet const& skip() noexcept
```

Comment:
Using a `static` and returning a `const&` instead of value may be a pessimization because of RVO, it's going to depend on how it's called. If it's used in a function call and bound to a `const&` paramter (`foo(skip())`) then this is an optimization. If it's used to initialize a value (`auto const a = skip()`) then it's a pessimization (there's extra coping because of missed RVO).

I think I'd keep this as `Keylet skip() noexcept`, but I'm OK if you want to keep the `const&`. I really just wanted to make sure you knew there was a trade-off here to consider. (same somment for `fees` as well)

Reply:
The thing is that these particular keylets are **actually** constants; they accept no arguments and their value is a fixed part of the protocol and will never change. From that perspective, I think that the return value being a `const&` is relevant.

However, you raise a good point that it may, actually, result in a pessimization. I'll take a look.